src/clj/clj_pdf/graphics_2d.clj

```diff
@@ -26,3 +32,54 @@
       (Rectangle. 0.0 0.0) ; choose a better placeholder?
       (finally
         (.dispose g2d)))))
+
+(def common-font-dirs
+  [["/Library/Fonts", true]
+   ["/System/Library/Fonts", false]
+   ["c:/windows/fonts", false]
+   ["c:/winnt/fonts", false]
+   ["d:/windows/fonts", false]
+   ["d:/winnt/fonts", false]
+   ["/usr/share/X11/fonts", true]
+   ["/usr/X/lib/X11/fonts", true]
+   ["/usr/openwin/lib/X11/fonts", true]
+   ["/usr/share/fonts", true]
+   ["/usr/X11R6/lib/X11/fonts", true]])
+
+(defn- full-path [parent filename]
+  (str/join [parent java.io.File/separator filename]))
+
+(defn g2d-register-fonts-helper [dir recursive]
+  (.insertDirectory default-font-mapper dir)
+  (if recursive
+    (let [fd (io/file dir)
+          files (.list fd)
+          paths (map #(full-path dir %) files)]
+      (doseq [path paths]
+        (if (.isDirectory (io/file path))
+          (g2d-register-fonts-helper
+           path recursive))))))
+
+(defn g2d-register-fonts
+  "Walks common font directories and registers them for use."
+  []
```

Comment:
would it make sense to allow passing optional custom dirs for fonts, something like:

```clojure
[& [font-dirs]]
...
(doseq [[dir recursive] (or font-dirs common-font-dirs)]
```